From d56ad081582f63666847daf622e4053f6882c075 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Thu, 2 Feb 2023 06:16:37 +0100
Subject: [PATCH] dnfpayload: make available repos for post-scripts

This is used by kickstart post-scripts to copy templates and
kernel-latest from installation source, whatever that source was.
---
 pyanaconda/payload/dnf/payload.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pyanaconda/payload/dnf/payload.py b/pyanaconda/payload/dnf/payload.py
index 2b0684a3a8..e5d8905221 100644
--- a/pyanaconda/payload/dnf/payload.py
+++ b/pyanaconda/payload/dnf/payload.py
@@ -446,6 +446,13 @@ class DNFPayload(Payload):
         log.debug("[Qubes OS]: Total templates RPMs size: %s", templates_size)
         return templates_size
 
+    def pre_install(self):
+        super().pre_install()
+        for repo in self._dnf_manager._base.repos.iter_enabled():
+            repo_file = '/tmp/installer.repo'
+            with open(repo_file, 'a') as repo_fd:
+                repo_fd.write(repo.dump())
+
     def install(self):
         self._progress_cb(0, _('Starting package installation process'))
 
-- 
2.39.0

